uaclient/entitlements/base.py

```diff
@@ -197,12 +197,14 @@ def applicability_status(self) -> "Tuple[ApplicabilityStatus, str]":
 
             if not match:
                 return ApplicabilityStatus.INAPPLICABLE, invalid_msg
-            if any(
-                [
-                    int(match.group("major")) < min_kern_major,
-                    int(match.group("minor")) < min_kern_minor,
-                ]
-            ):
+            kernel_major = int(match.group("major"))
+            kernel_minor = int(match.group("minor"))
+            major_gt_min = kernel_major > min_kern_major
+            minor_gte_min = bool(
+                kernel_major >= min_kern_major
+                and kernel_minor >= min_kern_minor
+            )
+            if not any([major_gt_min, minor_gte_min]):
```

Comment:
This line _reads_ strangely: "if either the major isn't greater than its minimum or the minor isn't greater than or equal to its minimum, then we are inapplicable". That's the previous logic we had, which is incorrect. I think changing `minor_gte_min` to a more accurate name would probably make this read a lot better, and make the code easier to grok.

(As a side note, I think this would be easier to grok in general if we had more than one return here, then we could do one condition at a time: `if kernel_major < min_kern_major: return ...` is clearer than defining a boolean which is referenced later in a compound if statement. This definitely doesn't need to happen in this PR or, indeed, at all.)

Reply:
Good suggestion Dan.  I like the early exit/return option better.  Fixed.